src/kv/test/kv_test.cpp

```diff
@@ -149,6 +161,126 @@ TEST_CASE("Reads/writes and deletions")
   }
 }
 
+TEST_CASE("get_previous_version")
```

Comment:
Can we also check that this is true across a primary and backup store and after deserialising snapshots?

Reply:
Modified the basic tests in `kv_serialisation.cpp` and `kv_snapshot.cpp` to test this - let me know if you think of any other awkward cases we should be testing.